File: backend/pkg/models/comment.go
package models

import "time"

// Comment represents a comment on a post
type Comment struct {
	ID        string    `json:"id"`
	PostID    string    `json:"post_id"`
	UserID    string    `json:"user_id"`
	Content   string    `json:"content"`
	ImageURL  string    `json:"image_url,omitempty"` // New field
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/types/Comment.ts
// frontend/src/types/Comment.ts
export interface Comment {
  id: string;
  post_id: string;
  user_id: string;
  content: string;
  image_url?: string;
  created_at: string; // ISO date string
  user_first_name?: string;
  user_last_name?: string;
  user_avatar_url?: string;
  username?: string;
}